pkg/apis/keycloak/v1alpha1/keycloak_types.go

```diff
@@ -51,6 +51,36 @@ type KeycloakSpec struct {
 	// Specify PodDisruptionBudget configuration
 	// +optional
 	PodDisruptionBudget PodDisruptionBudgetConfig `json:"podDisruptionBudget,omitempty"`
+	// Resources (Requests and Limits) for KeycloakDeployment and PostgresDeployment
+	// +optional
+	DeploymentSpec DeploymentSpec `json:"deploymentSpec,omitempty"`
+}
+
+type DeploymentSpec struct {
+	// Resources (Requests and Limits) for the KeycloakPods
+	// +optional
+	KeycloakDeploymentSpec KeycloakDeploymentSpec `json:"keycloakDeploymentSpec,omitempty"`
+	// Resources (Requests and Limits) for the PostgresPod
+	// +optional
+	PostgresDeploymentSpec PostgresDeploymentSpec `json:"postgresDeploymentSpec,omitempty"`
+}
+
+type KeycloakDeploymentSpec struct {
```

Comment:
I believe we could have a single struct for both Keycloak and Postgresql (comments and annotations omitted). Something like:

```
type DeploymentSpec struct {
	KeycloakDeploymentSpec ResourcesAndLimitsSpec `json:"keycloakDeploymentSpec,omitempty"`
	PostgresDeploymentSpec ResourcesAndLimitsSpec `json:"postgresDeploymentSpec,omitempty"`
}

type ResourcesAndLimitsSpec struct {
	Limits   ResourceList `json:"limits,omitempty"`
	Requests ResourceList `json:"requests,omitempty"`
}
```

The `ResourceList` might be taken from Kubernetes `v1` package (see` k8s.io/api/core/v1/types.go`). Implementing it this way, will save us some lines of code required for converting types from one to another.

Reply:
Single struct sounds good, this is probably too much. And using ResourceList-type from k8s sounds right too.

What do you think of this:
KeycloakSpec has now two separate entries for Keycloak- and Postgres-DeploymentSpec, so the structure is less deep.
Both entries have the same Type `DeploymentSpec` which contains the k8s-Type  `ResourceRequirements` with `Limits` and `Requests`.
If someone needs to specify more than just requests and limits, then `DeploymentSpec` can be extended


```
type KeycloakSpec struct {
...
	KeycloakDeploymentSpec DeploymentSpec `json:"keycloakDeploymentSpec,omitempty"`
	PostgresDeploymentSpec DeploymentSpec `json:"postgresDeploymentSpec,omitempty"`
}

type DeploymentSpec struct {
	// Resources (Requests and Limits) for the KeycloakPods
	// +optional
	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
}